modules/siddhi-extensions/time/src/main/java/org/wso2/siddhi/extension/time/ExtractDayOfWeekFunctionExtension.java

```diff
@@ -0,0 +1,170 @@
+ * dateValue - value of date. eg: "2014-11-11 13:23:44.657", "2014-11-11"
+ * dateFormat - Date format of the provided date value. eg: yyyy-MM-dd HH:mm:ss.SSS
+ * Accept Type(s) for dayOfWeek(dateValue,dateFormat):
+ * dateValue : STRING
+ * dateFormat : STRING
+ * Return Type(s): STRING
+ */
+public class ExtractDayOfWeekFunctionExtension extends FunctionExecutor {
+
+    private static final Logger log = Logger.getLogger(ExtractDayOfWeekFunctionExtension.class);
+    private Attribute.Type returnType = Attribute.Type.STRING;
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors,
+                        ExecutionPlanContext executionPlanContext) {
+        if (attributeExpressionExecutors.length > 2) {
+            throw new ExecutionPlanValidationException("Invalid no of arguments passed to time:dayOfWeek(dateValue," +
+                    "dateFormat) function, " + "required 2, but found " + attributeExpressionExecutors.length);
+        }
+        if (attributeExpressionExecutors[0].getReturnType() != Attribute.Type.STRING) {
+            throw new ExecutionPlanValidationException("Invalid parameter type found for the first argument of " +
+                    "time:dayOfWeek(dateValue,dateFormat) function, " + "required " + Attribute.Type.STRING +
+                    " but found " + attributeExpressionExecutors[0].getReturnType().toString());
+        }
+        //User can omit sending the dateFormat thus using a default CEP Time format
+        if (attributeExpressionExecutors.length > 0) {
+            if (attributeExpressionExecutors.length > 1 && attributeExpressionExecutors[1].getReturnType() != Attribute.Type.STRING) {
+                throw new ExecutionPlanValidationException("Invalid parameter type found for the second argument of " +
+                        "time:dayOfWeek(dateValue,dateFormat) function, " + "required " + Attribute.Type.STRING +
+                        " but found " + attributeExpressionExecutors[1].getReturnType().toString());
+            }
+        }
+
+    }
+
+    // this method will be executed for two or more parameters
+    @Override
+    protected Object execute(Object[] data) {
+        String userFormat;
+        if (data[0] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to time:dayOfWeek(dateValue," +
+                    "dateFormat) function" + ". First " + "argument cannot be null");
+        }
+        if (data.length > 1) {
+            if (data[1] == null) {
+                throw new ExecutionPlanRuntimeException(
+                        "Invalid input given to time:dayOfWeek(dateValue,dateFormat) function" + ". Second " +
+                                "argument cannot be null");
+            } else {
+                userFormat = (String) data[1];
+            }
+        } else {
+            userFormat = TimeExtensionConstants.EXTENSION_TIME_DEFAULT_DATE_FORMAT;
+        }
```

Comment:
@minudika 
Shall we add some spaces here to signify the logical separation validations and processing.. Add line breaks where it makes sense... Simply avoid putting any unnecessary line breaks...